### PR TITLE
Remove unneeded CSS vendor prefixes

### PR DESCRIPTION
**Pull request by adamzap**
- `box-sizing`: https://caniuse.com/css3-boxsizing
- `::placeholder`: https://caniuse.com/css-placeholder
- `::selection`: https://caniuse.com/css-selection
- `linear-gradient`: https://caniuse.com/css-gradients
- `appearance`: https://caniuse.com/css-appearance
- `user-select`: (partial): https://caniuse.com/user-select-none

I didn't remove vendor prefixes from `normalize.scss` or `font-awesome` assets because those libraries should probably be updated as a whole.

### CODE REVIEW

File: djangoproject/scss/_style.scss [hunk 1]
```diff
@@ -1209,8 +1205,6 @@ a.cta {
     // Call To Action buttons
     @include sans-serif;
     font-weight: 700;
-    -webkit-appearance: none;
-    -moz-appearance: none;
```
Comment (bmispelon) [inline djangoproject/scss/_style.scss:1213]: Shouldn't this get a non-prefixed replacement (`appearance: none`)?

**Review by bmispelon — COMMENTED** [review]
This looks great, and again the caniuse links were super helpful for doing the review 🙇🏻 

I noticed two cases where the original css only had prefixed versions and no standard one (`appearance`). I think we should add the non-prefixed version there, what do you think?

File: djangoproject/scss/_style.scss (hunk 1 above)
Comment (adamzap) [reply djangoproject/scss/_style.scss:1213]: You're right! Thanks, fixed!

File: djangoproject/scss/_style.scss [hunk 2]
```diff
@@ -2887,8 +2877,6 @@ form {
     textarea {
         @include sans-serif;
         @include font-size(16);
-        -webkit-appearance: none;
```
[the remark this answers is not in the document]
Comment (adamzap) [reply djangoproject/scss/_style.scss:2890]: Fixed!

**Review by bmispelon — APPROVED** [review]
Looking great 👍🏻

**Comment by bmispelon** [comment]
Merged and deployed, thanks again! 🚀

**Pull request merged**